<?xml version="1.0" encoding="UTF-8" ?>
<testsuite tests="54" failures="45" name="io.cloudboost.query.QueryTest" time="0.054" errors="9" skipped="0">
  <properties>
    <property name="java.runtime.name" value="Java(TM) SE Runtime Environment"/>
    <property name="sun.boot.library.path" value="C:\Program Files\jdk1.8.0_05\jre\bin"/>
    <property name="java.vm.version" value="25.5-b02"/>
    <property name="java.vm.vendor" value="Oracle Corporation"/>
    <property name="maven.multiModuleProjectDirectory" value="E:\Engineer\programming\projects\javaSDK\git_repo"/>
    <property name="java.vendor.url" value="http://java.oracle.com/"/>
    <property name="path.separator" value=";"/>
    <property name="guice.disable.misplaced.annotation.check" value="true"/>
    <property name="java.vm.name" value="Java HotSpot(TM) Client VM"/>
    <property name="file.encoding.pkg" value="sun.io"/>
    <property name="user.script" value=""/>
    <property name="user.country" value="US"/>
    <property name="sun.java.launcher" value="SUN_STANDARD"/>
    <property name="sun.os.patch.level" value=""/>
    <property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
    <property name="user.dir" value="E:\Engineer\programming\projects\javaSDK\git_repo"/>
    <property name="java.runtime.version" value="1.8.0_05-b13"/>
    <property name="java.awt.graphicsenv" value="sun.awt.Win32GraphicsEnvironment"/>
    <property name="java.endorsed.dirs" value="C:\Program Files\jdk1.8.0_05\jre\lib\endorsed"/>
    <property name="os.arch" value="x86"/>
    <property name="java.io.tmpdir" value="C:\Users\new\AppData\Local\Temp\"/>
    <property name="line.separator" value="
"/>
    <property name="java.vm.specification.vendor" value="Oracle Corporation"/>
    <property name="user.variant" value=""/>
    <property name="os.name" value="Windows 8.1"/>
    <property name="classworlds.conf" value="C:\Program Files (x86)\apache-maven-3.3.3\bin\..\bin\m2.conf"/>
    <property name="sun.jnu.encoding" value="Cp1252"/>
    <property name="java.library.path" value="C:\Program Files\jdk1.8.0_05\bin;C:\Windows\Sun\Java\bin;C:\Windows\system32;C:\Windows;C:\ProgramData\Oracle\Java\javapath;C:\oraclexe\app\oracle\product\11.2.0\server\bin;;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files (x86)\GNU\GnuPG\pub;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\nodejs\;C:\Program Files\MySQL\MySQL Server 5.6\bin;C:\Program Files (x86)\apache-ant-1.9.6\bin;C:\Program Files\jdk1.8.0_05\bin;E:\Engineer\eclipse\android-sdk-windows\tools;E:\Engineer\eclipse\android-sdk-windows\platform-tools;C:\Program Files\jdk1.8.0_05\bin;C:\Program Files\MongoDB\Server\3.0\bin;E:\Engineer\eclipse\eclipse;E:\Engineer\eclipse\myeclipse;C:\Program Files (x86)\apache-maven-3.3.3\bin;C:\Program Files (x86)\GNU\GnuPG;C:\Program Files\openssl-0.9.8k_X64\bin;C:\Program Files\nodejs;C:\Users\new\AppData\Roaming\npm;C:\Program Files\Wireshark;."/>
    <property name="java.specification.name" value="Java Platform API Specification"/>
    <property name="java.class.version" value="52.0"/>
    <property name="sun.management.compiler" value="HotSpot Client Compiler"/>
    <property name="os.version" value="6.3"/>
    <property name="user.home" value="C:\Users\new"/>
    <property name="user.timezone" value="Africa/Nairobi"/>
    <property name="java.awt.printerjob" value="sun.awt.windows.WPrinterJob"/>
    <property name="java.specification.version" value="1.8"/>
    <property name="file.encoding" value="Cp1252"/>
    <property name="user.name" value="new"/>
    <property name="java.class.path" value="C:\Program Files (x86)\apache-maven-3.3.3\bin\..\boot\plexus-classworlds-2.5.2.jar"/>
    <property name="java.vm.specification.version" value="1.8"/>
    <property name="sun.arch.data.model" value="32"/>
    <property name="java.home" value="C:\Program Files\jdk1.8.0_05\jre"/>
    <property name="sun.java.command" value="org.codehaus.plexus.classworlds.launcher.Launcher test"/>
    <property name="java.specification.vendor" value="Oracle Corporation"/>
    <property name="user.language" value="en"/>
    <property name="awt.toolkit" value="sun.awt.windows.WToolkit"/>
    <property name="java.vm.info" value="mixed mode"/>
    <property name="java.version" value="1.8.0_05"/>
    <property name="java.ext.dirs" value="C:\Program Files\jdk1.8.0_05\jre\lib\ext;C:\Windows\Sun\Java\lib\ext"/>
    <property name="sun.boot.class.path" value="C:\Program Files\jdk1.8.0_05\jre\lib\resources.jar;C:\Program Files\jdk1.8.0_05\jre\lib\rt.jar;C:\Program Files\jdk1.8.0_05\jre\lib\sunrsasign.jar;C:\Program Files\jdk1.8.0_05\jre\lib\jsse.jar;C:\Program Files\jdk1.8.0_05\jre\lib\jce.jar;C:\Program Files\jdk1.8.0_05\jre\lib\charsets.jar;C:\Program Files\jdk1.8.0_05\jre\lib\jfr.jar;C:\Program Files\jdk1.8.0_05\jre\classes"/>
    <property name="sun.stderr.encoding" value="cp437"/>
    <property name="java.vendor" value="Oracle Corporation"/>
    <property name="maven.home" value="C:\Program Files (x86)\apache-maven-3.3.3\bin\.."/>
    <property name="file.separator" value="\"/>
    <property name="java.vendor.url.bug" value="http://bugreport.sun.com/bugreport/"/>
    <property name="sun.cpu.endian" value="little"/>
    <property name="sun.io.unicode.encoding" value="UnicodeLittle"/>
    <property name="sun.stdout.encoding" value="cp437"/>
    <property name="sun.desktop" value="windows"/>
    <property name="sun.cpu.isalist" value="pentium_pro+mmx pentium_pro pentium+mmx pentium i486 i386 i86"/>
  </properties>
  <testcase classname="io.cloudboost.query.QueryTest" name="startsWith" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$22.done(QueryTest.java:636)
	at io.cloudboost.CloudObject.saveAll(CloudObject.java:981)
	at io.cloudboost.query.QueryTest.startsWith(QueryTest.java:631)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="containsAll" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$21.done(QueryTest.java:582)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.containsAll(QueryTest.java:577)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="exists" time="0.001">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$31$1.done(QueryTest.java:927)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$31.done(QueryTest.java:924)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.exists(QueryTest.java:919)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$31$1.done(QueryTest.java:927)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$31.done(QueryTest.java:924)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.exists(QueryTest.java:919)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="greaterThan" time="0.001">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$25$1.done(QueryTest.java:747)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$25.done(QueryTest.java:744)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.greaterThan(QueryTest.java:739)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$25$1.done(QueryTest.java:747)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$25.done(QueryTest.java:744)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.greaterThan(QueryTest.java:739)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="doesNotExists" time="0">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$32$1.done(QueryTest.java:955)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$32.done(QueryTest.java:952)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.doesNotExists(QueryTest.java:947)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$32$1.done(QueryTest.java:955)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$32.done(QueryTest.java:952)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.doesNotExists(QueryTest.java:947)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="lessThanEqualTo" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$28.done(QueryTest.java:829)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.lessThanEqualTo(QueryTest.java:825)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="notContianedIn" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$20.done(QueryTest.java:542)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest.notContianedIn(QueryTest.java:537)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="ascendingOrder" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$29.done(QueryTest.java:859)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.ascendingOrder(QueryTest.java:855)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveRelation" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$33.done(QueryTest.java:984)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveRelation(QueryTest.java:980)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="descendingOrder" time="0.001">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$30$1.done(QueryTest.java:895)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$30.done(QueryTest.java:892)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.descendingOrder(QueryTest.java:887)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$30$1.done(QueryTest.java:895)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$30.done(QueryTest.java:892)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.descendingOrder(QueryTest.java:887)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveMultiJoin" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$34.done(QueryTest.java:1007)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveMultiJoin(QueryTest.java:1003)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="includeFindById" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$35.done(QueryTest.java:1025)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.includeFindById(QueryTest.java:1021)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveListInColumn" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at junit.framework.Assert.fail(Assert.java:53)
	at io.cloudboost.query.QueryTest$6.done(QueryTest.java:196)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveListInColumn(QueryTest.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="selectColumn" time="0.001">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$11.done(QueryTest.java:291)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.selectColumn(QueryTest.java:287)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$11.done(QueryTest.java:291)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.selectColumn(QueryTest.java:287)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="equalToWithNull" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$14$1.done(QueryTest.java:373)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$14.done(QueryTest.java:369)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.equalToWithNull(QueryTest.java:364)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="findItemById" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$17.done(QueryTest.java:450)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.findItemById(QueryTest.java:445)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="findDataWithId" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$18.done(QueryTest.java:484)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.findDataWithId(QueryTest.java:479)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldNotRetrieveDataWithAparticularValue" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$42.done(QueryTest.java:1291)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest.shouldNotRetrieveDataWithAparticularValue(QueryTest.java:1285)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="notEqualToWithNull" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$15$1.done(QueryTest.java:402)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$15.done(QueryTest.java:398)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.notEqualToWithNull(QueryTest.java:393)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveListwithInColumn" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at junit.framework.Assert.fail(Assert.java:53)
	at io.cloudboost.query.QueryTest$10.done(QueryTest.java:274)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveListwithInColumn(QueryTest.java:269)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="getEncryptedPassword" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$38$1.done(QueryTest.java:1144)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$38.done(QueryTest.java:1140)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.getEncryptedPassword(QueryTest.java:1134)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveDataWithGivenValue" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$8.done(QueryTest.java:227)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveDataWithGivenValue(QueryTest.java:222)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="selectColumnDistinct" time="0.002">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$12.done(QueryTest.java:322)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.selectColumnDistinct(QueryTest.java:318)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$12.done(QueryTest.java:322)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.selectColumnDistinct(QueryTest.java:318)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveListWithinColumn" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at junit.framework.Assert.fail(Assert.java:53)
	at io.cloudboost.query.QueryTest$54.done(QueryTest.java:1729)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveListWithinColumn(QueryTest.java:1724)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="greaterThanEqualTo" time="0.003">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$26$1.done(QueryTest.java:776)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$26.done(QueryTest.java:772)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.greaterThanEqualTo(QueryTest.java:767)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="findOne" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$16.done(QueryTest.java:428)
	at io.cloudboost.CloudQuery.findOne(CloudQuery.java:1516)
	at io.cloudboost.query.QueryTest.findOne(QueryTest.java:424)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="findData" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$19.done(QueryTest.java:517)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest.findData(QueryTest.java:513)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="findById" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$23.done(QueryTest.java:673)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.findById(QueryTest.java:668)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="lessThan" time="0.003">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$27$1.done(QueryTest.java:805)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$27.done(QueryTest.java:802)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.lessThan(QueryTest.java:797)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$27$1.done(QueryTest.java:805)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest$27.done(QueryTest.java:802)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.lessThan(QueryTest.java:797)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldRetrieveWhenNotNullColumn" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$1.done(QueryTest.java:35)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldRetrieveWhenNotNullColumn(QueryTest.java:30)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldRetrieveWhenNullColumn" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$3.done(QueryTest.java:97)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldRetrieveWhenNullColumn(QueryTest.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="getEncryptedPasswordOverOr" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$36.done(QueryTest.java:1059)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.getEncryptedPasswordOverOr(QueryTest.java:1054)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldQueryOverLinkedColumn" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$50.done(QueryTest.java:1581)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldQueryOverLinkedColumn(QueryTest.java:1575)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="selectColumnShouldWorkOnDistinct" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$4.done(QueryTest.java:131)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.selectColumnShouldWorkOnDistinct(QueryTest.java:126)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldReturnCountAsInteger" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$41.done(QueryTest.java:1264)
	at io.cloudboost.CloudQuery.count(CloudQuery.java:1282)
	at io.cloudboost.query.QueryTest.shouldReturnCountAsInteger(QueryTest.java:1259)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="queryOverLinkedColumnWithEqualTo" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$24.done(QueryTest.java:711)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.queryOverLinkedColumnWithEqualTo(QueryTest.java:707)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldLimitNumberOfDataItems" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$43.done(QueryTest.java:1317)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest.shouldLimitNumberOfDataItems(QueryTest.java:1312)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="selectColumnShouldWorkOnFind" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$5.done(QueryTest.java:164)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.selectColumnShouldWorkOnFind(QueryTest.java:159)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveDataWithAParticularValue" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$13.done(QueryTest.java:353)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveDataWithAParticularValue(QueryTest.java:348)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldIncludeRelationOnDistinct" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$49.done(QueryTest.java:1530)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldIncludeRelationOnDistinct(QueryTest.java:1524)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldQueryOverBooleanType" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$47.done(QueryTest.java:1428)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldQueryOverBooleanType(QueryTest.java:1423)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="retrieveDataWithParticularValue" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at junit.framework.Assert.fail(Assert.java:53)
	at io.cloudboost.query.QueryTest$9.done(QueryTest.java:243)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.retrieveDataWithParticularValue(QueryTest.java:238)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="retrieveElementWithGivenId" time="0.003">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$53.done(QueryTest.java:1694)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.retrieveElementWithGivenId(QueryTest.java:1689)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="getEncryptedPasswordOverQuery" time="0.002">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$39.done(QueryTest.java:1168)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.getEncryptedPasswordOverQuery(QueryTest.java:1165)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$39.done(QueryTest.java:1168)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.getEncryptedPasswordOverQuery(QueryTest.java:1165)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldIncludeWithFindById" time="0">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$52.done(QueryTest.java:1661)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldIncludeWithFindById(QueryTest.java:1655)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="containedInShouldWorkOnId" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$40.done(QueryTest.java:1205)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.containedInShouldWorkOnId(QueryTest.java:1200)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="containedInOverListOfCloudObject" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$51.done(QueryTest.java:1618)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.containedInOverListOfCloudObject(QueryTest.java:1613)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="saveDataWithParticularValue" time="0.001">
    <error type="java.lang.NullPointerException">java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$7.done(QueryTest.java:212)
	at io.cloudboost.CloudObject.save(CloudObject.java:890)
	at io.cloudboost.query.QueryTest.saveDataWithParticularValue(QueryTest.java:208)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</error>
    <system-err>java.lang.NullPointerException
	at io.cloudboost.query.QueryTest$7.done(QueryTest.java:212)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.saveDataWithParticularValue(QueryTest.java:208)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</system-err>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldRetrieveWhenNotNullColumnNotEqFunc" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$2.done(QueryTest.java:67)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldRetrieveWhenNotNullColumnNotEqFunc(QueryTest.java:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldNotEncryptAlreadyEncryptedPassword" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$37.done(QueryTest.java:1100)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldNotEncryptAlreadyEncryptedPassword(QueryTest.java:1095)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldLimitNumberOfDataItemsToOne" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at junit.framework.Assert.assertTrue(Assert.java:20)
	at junit.framework.Assert.assertTrue(Assert.java:27)
	at io.cloudboost.query.QueryTest$44.done(QueryTest.java:1334)
	at io.cloudboost.CloudQuery.findOne(CloudQuery.java:1516)
	at io.cloudboost.query.QueryTest.shouldLimitNumberOfDataItemsToOne(QueryTest.java:1330)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldGetElementNotHavingGivenColumnName" time="0.001">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$45.done(QueryTest.java:1350)
	at io.cloudboost.CloudQuery.find(CloudQuery.java:1425)
	at io.cloudboost.query.QueryTest.shouldGetElementNotHavingGivenColumnName(QueryTest.java:1345)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldNotGiveElementWithGivenRelation" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$46.done(QueryTest.java:1374)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldNotGiveElementWithGivenRelation(QueryTest.java:1368)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
  <testcase classname="io.cloudboost.query.QueryTest" name="shouldIncludeRelationObjectWhenIncludeIsRequestedInQuery" time="0.002">
    <failure type="junit.framework.AssertionFailedError:">junit.framework.AssertionFailedError: null
	at junit.framework.Assert.fail(Assert.java:47)
	at io.cloudboost.query.QueryTest$48.done(QueryTest.java:1475)
	at io.cloudboost.CloudObject.save(CloudObject.java:885)
	at io.cloudboost.query.QueryTest.shouldIncludeRelationObjectWhenIncludeIsRequestedInQuery(QueryTest.java:1469)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:44)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:41)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$1.run(FailOnTimeout.java:28)
</failure>
  </testcase>
</testsuite>